.travis.yml

```diff
@@ -114,45 +105,66 @@ before_install:
           if [[ -e $ext_cache/$ext_so ]]; then
               echo extension = $ext_cache/$ext_so >> $INI
           else
+              rm ~/.pearrc /tmp/pear 2>/dev/null || true
               mkdir -p $ext_cache
               echo yes | pecl install -f $ext_name &&
               cp $ext_dir/$ext_so $ext_cache
           fi
       }
       export -f tpecl
 
-      # Matrix lines for intermediate PHP versions are skipped for pull requests
-      if [[ ! $deps && ! $PHP = ${MIN_PHP%.*} && ! $PHP = hhvm* && $TRAVIS_PULL_REQUEST != false ]]; then
-          deps=skip
-          skip=1
-      else
-          COMPONENTS=$(find src/Symfony -mindepth 3 -type f -name phpunit.xml.dist -printf '%h\n')
-      fi
-
     - |
       # Install sigchild-enabled PHP to test the Process component on the lowest PHP matrix line
-      if [[ ! $deps && $PHP = ${MIN_PHP%.*} && ! -d php-$MIN_PHP/sapi ]]; then
+      if [[ ! $deps && $TRAVIS_PHP_VERSION = ${MIN_PHP%.*} && ! -d php-$MIN_PHP/sapi ]]; then
           wget http://museum.php.net/php5/php-$MIN_PHP.tar.bz2 -O - | tar -xj &&
           (cd php-$MIN_PHP && ./configure --enable-sigchild --enable-pcntl && make -j2)
       fi
 
+    - |
+      # php.ini configuration
+      for PHP in $TRAVIS_PHP_VERSION $php_extra; do
+          if [[ $PHP = hhvm* ]]; then
+              INI=/etc/hhvm/php.ini
+          else
+              phpenv global $PHP 2>/dev/null || (cd / && wget https://s3.amazonaws.com/travis-php-archives/binaries/ubuntu/14.04/x86_64/php-$PHP.tar.bz2 -O - | tar -xj)
+              INI=~/.phpenv/versions/$PHP/etc/conf.d/travis.ini
+          fi
+          echo date.timezone = Europe/Paris >> $INI
+          echo memory_limit = -1 >> $INI
+          echo session.gc_probability = 0 >> $INI
+          echo opcache.enable_cli = 1 >> $INI
+          echo hhvm.jit = 0 >> $INI
+          echo apc.enable_cli = 1 >> $INI
+          [[ $PHP = 5.* ]] && echo extension = memcache.so >> $INI
+          if [[ $PHP = 5.* ]]; then
+              echo extension = mongo.so >> $INI
+          fi
+      done
+
     - |
       # Install extra PHP extensions
-      if [[ ! $skip && $PHP = 5.* ]]; then
-          ([[ $deps ]] || tfold ext.symfony_debug 'cd src/Symfony/Component/Debug/Resources/ext && phpize && ./configure && make && echo extension = $(pwd)/modules/symfony_debug.so >> '"$INI")
-          tfold ext.memcached tpecl memcached-2.1.0 memcached.so $INI
-          tfold ext.apcu tpecl apcu-4.0.11 apcu.so $INI
-      elif [[ ! $skip && $PHP = 7.* ]]; then
-          tfold ext.apcu tpecl apcu-5.1.6 apcu.so $INI
-          tfold ext.mongodb tpecl mongodb-1.5.0 mongodb.so $INI
-      fi
+      for PHP in $TRAVIS_PHP_VERSION $php_extra; do
```

Comment:
if we skip extra versions, should we still install extensions for them ?

Reply:
The compilation is cached so skipping is not required (makes the logic a bit simpler here.)